<br><p style="font-family: Arial; font-size:3.75em;color:purple; font-style:bold">
Matplotlib: Exploring <br> <br> <br>Data Visualization</p><br><br>

<br><br><center><h1 style="font-size:2em;color:#2467C0">World Development Indicators</h1></center>
<br>
<table>
<col width="550">
<col width="450">
<tr>
<td><img src="https://upload.wikimedia.org/wikipedia/commons/4/46/North_South_divide.svg" align="middle" style="width:550px;height:360px;"/></td>
<td>
This week, we will be using an open dataset from <a href="https://www.kaggle.com">Kaggle</a>. It is  <a href="https://www.kaggle.com/worldbank/world-development-indicators">The World Development Indicators</a> dataset obtained from the World Bank containing over a thousand annual indicators of economic development from hundreds of countries around the world.
<br>
<br>
This is a slightly modified version of the original dataset from <a href="http://data.worldbank.org/data-catalog/world-development-indicators">The World Bank</a>
<br>
<br>
List of the <a href="https://www.kaggle.com/benhamner/d/worldbank/world-development-indicators/indicators-in-data">available indicators</a> and a <a href="https://www.kaggle.com/benhamner/d/worldbank/world-development-indicators/countries-in-the-wdi-data">list of the available countries</a>.
</td>
</tr>
</table>

# Step1: Initial exploration of the DataSet

In [1]:
import pandas as pd
import random 
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

In [2]:
ls

 Volume in drive C has no label.
 Volume Serial Number is 372F-3690

 Directory of C:\Users\USER\Documents\Python for data science\Week5-Visualization

10/23/2017  03:15 PM    <DIR>          .
10/23/2017  03:15 PM    <DIR>          ..
10/23/2017  03:12 PM    <DIR>          .ipynb_checkpoints
10/23/2017  03:14 PM           124,661 05a_Matplotlib_Notebook.ipynb
05/26/2017  04:51 PM           511,733 05b_Exploring Indicator's Across Countries.ipynb
05/26/2017  04:51 PM            14,054 05c_Folium_Notebook.ipynb
05/26/2017  02:38 PM                 0 Icon_
10/23/2017  03:03 PM         7,550,061 Indicators.csv
10/23/2017  03:00 PM        62,169,137 WDI_csv.zip
10/19/2017  10:38 AM           147,461 WDICountry.csv
10/19/2017  10:38 AM           822,861 WDICountry-Series.csv
10/19/2017  10:38 AM       200,398,556 WDIData.csv
10/19/2017  10:39 AM        48,817,914 WDIFootNote.csv
10/19/2017  10:38 AM         3,787,335 WDISeries.csv
10/19/2017  10:38 AM            37,203 WDISeries-Time.csv
10/

In [3]:
data = pd.read_csv("Indicators.csv")

In [4]:
data.shape

(5656458, 6)

What the fuck is it? Nearly 6M rows =))

In [5]:
data.head()

CountryName CountryCode                                      IndicatorName  \
0  Arab World         ARB  Adolescent fertility rate (births per 1,000 wo...   
1  Arab World         ARB  Age dependency ratio (% of working-age populat...   
2  Arab World         ARB  Age dependency ratio, old (% of working-age po...   
3  Arab World         ARB  Age dependency ratio, young (% of working-age ...   
4  Arab World         ARB        Arms exports (SIPRI trend indicator values)   

    IndicatorCode  Year         Value  
0     SP.ADO.TFRT  1960  1.335609e+02  
1     SP.POP.DPND  1960  8.779760e+01  
2  SP.POP.DPND.OL  1960  6.634579e+00  
3  SP.POP.DPND.YG  1960  8.102333e+01  
4  MS.MIL.XPRT.KD  1960  3.000000e+06

# How many UNIQUE country names are there?

In [25]:
countries = data["CountryName"].unique().tolist()
countries 

['Arab World',
 'Caribbean small states',
 'Central Europe and the Baltics',
 'East Asia & Pacific (all income levels)',
 'East Asia & Pacific (developing only)',
 'Euro area',
 'Europe & Central Asia (all income levels)',
 'Europe & Central Asia (developing only)',
 'European Union',
 'Fragile and conflict affected situations',
 'Heavily indebted poor countries (HIPC)',
 'High income',
 'High income: nonOECD',
 'High income: OECD',
 'Latin America & Caribbean (all income levels)',
 'Latin America & Caribbean (developing only)',
 'Least developed countries: UN classification',
 'Low & middle income',
 'Low income',
 'Lower middle income',
 'Middle East & North Africa (all income levels)',
 'Middle East & North Africa (developing only)',
 'Middle income',
 'North America',
 'OECD members',
 'Other small states',
 'Pacific island small states',
 'Small states',
 'South Asia',
 'Sub-Saharan Africa (all income levels)',
 'Sub-Saharan Africa (developing only)',
 'Upper middle income',
 'Wor

In [26]:
len(countries)

247

=) There are 247 Nations!

# Are there the same number of country code?

In [27]:
countries_code =data["CountryCode"].unique().tolist()
len(countries_code)

247

Okay, here are 247 rows as well

# Are there many Indicators or just a few?

In [30]:
indicators = data["IndicatorName"].unique().tolist()
len(indicators)

1344

# LEt's look at How Vietname is


In [23]:
data[data.CountryName == "Vietnam"]

CountryName CountryCode  \
22799       Vietnam         VNM   
22800       Vietnam         VNM   
22801       Vietnam         VNM   
22802       Vietnam         VNM   
22803       Vietnam         VNM   
22804       Vietnam         VNM   
22805       Vietnam         VNM   
22806       Vietnam         VNM   
22807       Vietnam         VNM   
22808       Vietnam         VNM   
22809       Vietnam         VNM   
22810       Vietnam         VNM   
22811       Vietnam         VNM   
22812       Vietnam         VNM   
22813       Vietnam         VNM   
22814       Vietnam         VNM   
22815       Vietnam         VNM   
22816       Vietnam         VNM   
22817       Vietnam         VNM   
22818       Vietnam         VNM   
22819       Vietnam         VNM   
22820       Vietnam         VNM   
22821       Vietnam         VNM   
22822       Vietnam         VNM   
22823       Vietnam         VNM   
22824       Vietnam         VNM   
22825       Vietnam         VNM   
22826       Vietnam         VNM   
22827       Vietnam         VNM   
22828       Vietnam         VNM   
...             ...         ...   
5656141     Vietnam         VNM   
5656142     Vietnam         VNM   
5656143     Vietnam         VNM   
5656144     Vietnam         VNM   
5656145     Vietnam         VNM   
5656146     Vietnam         VNM   
5656147     Vietnam         VNM   
5656148     Vietnam         VNM   
5656149     Vietnam         VNM   
5656150     Vietnam         VNM   
5656151     Vietnam         VNM   
5656152     Vietnam         VNM   
5656153     Vietnam         VNM   
5656154     Vietnam         VNM   
5656155     Vietnam         VNM   
5656156     Vietnam         VNM   
5656157     Vietnam         VNM   
5656158     Vietnam         VNM   
5656159     Vietnam         VNM   
5656160     Vietnam         VNM   
5656161     Vietnam         VNM   
5656162     Vietnam         VNM   
5656163     Vietnam         VNM   
5656164     Vietnam         VNM   
5656165     Vietnam         VNM   
5656166     Vietnam         VNM   
5656167     Vietnam         VNM   
5656168     Vietnam         VNM   
5656169     Vietnam         VNM   
5656170     Vietnam         VNM   

                                             IndicatorName      IndicatorCode  \
22799    Adolescent fertility rate (births per 1,000 wo...        SP.ADO.TFRT   
22800    Age dependency ratio (% of working-age populat...        SP.POP.DPND   
22801    Age dependency ratio, old (% of working-age po...     SP.POP.DPND.OL   
22802    Age dependency ratio, young (% of working-age ...     SP.POP.DPND.YG   
22803          Arms imports (SIPRI trend indicator values)     MS.MIL.MPRT.KD   
22804                 Birth rate, crude (per 1,000 people)     SP.DYN.CBRT.IN   
22805                                   CO2 emissions (kt)     EN.ATM.CO2E.KT   
22806               CO2 emissions (metric tons per capita)     EN.ATM.CO2E.PC   
22807    CO2 emissions from gaseous fuel consumption (%...  EN.ATM.CO2E.GF.ZS   
22808     CO2 emissions from gaseous fuel consumption (kt)  EN.ATM.CO2E.GF.KT   
22809    CO2 emissions from liquid fuel consumption (% ...  EN.ATM.CO2E.LF.ZS   
22810      CO2 emissions from liquid fuel consumption (kt)  EN.ATM.CO2E.LF.KT   
22811    CO2 emissions from solid fuel consumption (% o...  EN.ATM.CO2E.SF.ZS   
22812       CO2 emissions from solid fuel consumption (kt)  EN.ATM.CO2E.SF.KT   
22813                 Death rate, crude (per 1,000 people)     SP.DYN.CDRT.IN   
22814             Fertility rate, total (births per woman)     SP.DYN.TFRT.IN   
22815                        Fixed telephone subscriptions        IT.MLT.MAIN   
22816       Fixed telephone subscriptions (per 100 people)     IT.MLT.MAIN.P2   
22817        International migrant stock (% of population)     SM.POP.TOTL.ZS   
22818                   International migrant stock, total        SM.POP.TOTL   
22819             Life expectancy at birth, female (years)  SP.DYN.LE00.FE.IN   
22820               Life expectancy at birth, male (yea

# How many years of data do we have?

In [32]:
years = data["Year"].unique().tolist()
len(years)

56

# What is the range of years?

In [36]:
print ( min(years) , "to" , max(years))

1960 to 2015
